Runtimes/Overlay/cmake/modules/CMakeWorkarounds.cmake

```diff
@@ -0,0 +1,34 @@
+# This file contains workarounds for dealing with bugs between CMake and the
+# Swift compiler.
+
+# This is a workaround to avoid crashing the Swift compiler while typechecking
+# swiftCore. The Swift compiler must be run num-threads=0 with WMO in all build
+# configurations.
+# These must be set in CMake _before_ the Swift language is enabled.
+# This is manipulating undocumented CMake-internal variables that may change
+# behavior at any time and should not be relied on.
+# Otherwise CMake will generate the build graph incorrectly for this setup.
+# Ideally this file should not need to exist.
+
+set(CMAKE_Swift_NUM_THREADS 0)
```

Comment:
I don't think we need this file for the overlays. This file is required in `Core` because `libswiftCore` has to be built with `-wmo -num-threads 0`, but there isn't a mechanism to specify `-num-threads 0` through normal CMake at the moment.